lisa/transformers/scripts_execution_transformer.py

```diff
@@ -0,0 +1,95 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+from dataclasses import dataclass, field
+from enum import Enum
+from typing import Any, Dict, List, Optional, Type
+
+from dataclasses_json import dataclass_json
+
+from lisa import schema
+from lisa.transformers.deployment_transformer import (
+    DeploymentTransformer,
+    DeploymentTransformerSchema,
+)
+from lisa.util import LisaException
+
+
+class ScriptInterpreter(str, Enum):
+    BASH = "bash"
+
+
+@dataclass_json()
+@dataclass
+class ScriptEntry:
+    script: str = ""
+    interpreter: ScriptInterpreter = ScriptInterpreter.BASH
+    args: Optional[str] = None
+
+
+@dataclass_json()
+@dataclass
+class ScriptsExecutionTransformerSchema(DeploymentTransformerSchema):
+    dependent_packages: List[str] = field(default_factory=list)
+    scripts: List[ScriptEntry] = field(default_factory=list)
+    reboot: bool = field(default=False)
+    exit_on_error: bool = field(default=False)
```

Comment:
See above comments, with the `expected_exit_code`, this field is not needed.

Reply:
updated